From 9a1b0695277a3864b42d082095962f8742cdcf04 Mon Sep 17 00:00:00 2001
From: Denis Dzyubenko <denis.dzyubenko@nokia.com>
Date: Thu, 1 Jul 2010 12:00:05 +0200
Subject: [PATCH] Fixed QX11EmbedContainer.

Enforce the creation of the native window id for the focusproxy widget inside
the container to make sure we won't get a CreateNotify event to try to embed
the focusproxy itself.

Task-number: QTBUG-10809
Reviewed-by: Bradley T. Hughes
---
 src/gui/kernel/qx11embed_x11.cpp |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/src/gui/kernel/qx11embed_x11.cpp b/src/gui/kernel/qx11embed_x11.cpp
index b527e72..9f1b1f8 100644
--- a/src/gui/kernel/qx11embed_x11.cpp
+++ b/src/gui/kernel/qx11embed_x11.cpp
@@ -1070,6 +1070,7 @@ QX11EmbedContainer::QX11EmbedContainer(QWidget *parent)
     d->focusProxy = new QWidget(this);
     d->focusProxy->setAttribute(Qt::WA_NativeWindow);
     d->focusProxy->setAttribute(Qt::WA_DontCreateNativeAncestors);
+    d->focusProxy->createWinId();
     d->focusProxy->setGeometry(-1, -1, 1, 1);
 
     // We need events from the window (activation status) and
-- 
1.6.1

